core/corehttp/gateway_handler_unixfs_dir.go

```diff
@@ -220,3 +202,36 @@ func (i *gatewayHandler) serveDirectory(ctx context.Context, w http.ResponseWrit
 func getDirListingEtag(dirCid cid.Cid) string {
 	return `"DirIndex-` + assets.AssetHash + `_CID-` + dirCid.String() + `"`
 }
+
+func (i *gatewayHandler) getUnixFsLinks(ctx context.Context, cid cid.Cid) ([]*format.Link, error) {
```

Comment:
It does what this code does, plus much more, that we do not need. That was my idea.

Reply:
UnixFS.LS should give the file size, not the cumulative size IMO. that's one of the differences.

Reply:
> to ensure we have the correct file size

It doesn't really, it only fetch the top most node.

Sounds like you could add an option to `.Ls` to use cumulative size instead of fetching the top most node, and avoid most of the duplicated code here.

Reply:
Updated.